src/services/artist.service.ts

```diff
@@ -0,0 +1,75 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { IArtist } from 'src/Interfaces/IArtist';
+import { Observable, throwError, timer } from 'rxjs';
+import {
+  catchError,
+  tap,
+  map,
+  distinctUntilChanged,
+  retry
+} from 'rxjs/operators';
+import { IAlbum } from 'src/Interfaces/iAlbum';
+import { ITrack } from 'src/Interfaces/iTrack';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ArtistService {
+  private ArtistListUrl: string;
+  private artistUrl: string;
+  private albumsUrl: string;
+  private trackUrl: string;
+  constructor(private http: HttpClient) {}
+
+  getArtistList(str: string): Observable<IArtist[]> {
+    this.ArtistListUrl =
+      'https://cors-anywhere.herokuapp.com/https://api.deezer.com/search/artist?q=' +
```

Comment:
Consider storing this URL inside a const variable in `ArtistService` or `environment.ts` and `environment.prod.ts` files so that it is reusable.

Reply:
@strizzwald fixed